### PR TITLE
[master] Some unit test fixes

### LINKED ISSUE #622: Update xUnit version reference in master
Right now we're using xunit 2.1.0 (released) in xplat, but `2.1.0-rc1-build3168` in master. We should go to the RTM version.

### PR DESCRIPTION
**Pull request by AndyGerlicher**
This was very painful.

Down to 24 skipped now and the rest passing (on my machine anyway, hopefully the CI agrees). Most of the skipped now were skipped in our internal sources as well (some are just broken scenarios that I have no idea if they should be fixed).

Also closes #622.

### CODE REVIEW

**Comment by AndyGerlicher** [comment]
@dotnet-bot test this please

File: src/XMakeBuildEngine/UnitTests/BackEnd/TaskRegistry_Tests.cs [hunk 1]
```diff
@@ -717,20 +713,24 @@ public void RetrieveFromCacheMatchingExactParameters_AdditionalParameters()
...
         /// fuzzy matches.
         /// </summary>
-        [Fact(Skip = "Test fails in xunit when multiple tests are run")]
+        [Fact]
         public void RetrieveFromCacheFuzzyMatchingParameters()
         {
-            Assert.NotNull(s_testTaskLocation); // "Need a test task to run this test"
+            if (_testTaskLocation == null)
+            {
+                Debugger.Break();
```
Comment (rainersigwald) [inline src/XMakeBuildEngine/UnitTests/BackEnd/TaskRegistry_Tests.cs]: Looks like this should be removed.

File: src/XMakeBuildEngine/UnitTests/BackEnd/TaskRegistry_Tests.cs [hunk 2]
```diff
@@ -1040,20 +1040,24 @@ public void RetrieveFromCacheFuzzyMatchingParameters_MultipleUsingTasks_PreferCa
...
         /// fuzzy matches.
         /// </summary>
-        [Fact(Skip = "Test fails in xunit when multiple tests are run")]
+        [Fact]
         public void RetrieveFromCacheFuzzyMatchingParameters_ExactMatches()
         {
-            Assert.NotNull(s_testTaskLocation); // "Need a test task to run this test"
+            if (_testTaskLocation == null)
+            {
+                Debugger.Break();
```
Comment (rainersigwald) [inline src/XMakeBuildEngine/UnitTests/BackEnd/TaskRegistry_Tests.cs]: ditto

File: src/XMakeBuildEngine/UnitTests/BackEnd/TaskRegistry_Tests.cs [hunk 3]
```diff
@@ -2063,18 +2073,109 @@ private static List<ProjectUsingTaskElement> CreateTaskBodyElementWithAttributes
...
+        /// </summary>
+        internal static int GetDeepCountOfRegisteredTasks(IDictionary<TaskRegistry.RegisteredTaskIdentity, List<TaskRegistry.RegisteredTaskRecord>> registryRecords)
+        {
+            return registryRecords?.Values.Sum(recordList => recordList.Count) ?? 0;
+        }
+
+        /// <summary>
+        /// Create and fill a task registry based on some using task elements.
+        /// </summary>
+        internal TaskRegistry CreateTaskRegistryAndRegisterTasks(List<ProjectUsingTaskElement> usingTaskElement)
```
Comment (rainersigwald) [inline src/XMakeBuildEngine/UnitTests/BackEnd/TaskRegistry_Tests.cs]: Shouldn't the list stay plural?

File: src/Utilities/UnitTests/ToolLocationHelper_Tests.cs [hunk 4]
```diff
@@ -1082,7 +1082,7 @@ public void VerifyToolsetAndToolLocationHelperAgreeWhenVisualStudioVersionIs10()
         }
 
 #if STANDALONEBUILD
-        [Fact(Skip = "Test failed in MSTest pre-xunit conversion")]
+        [Fact]
 #else
         [Fact]
 #endif
```
Comment (rainersigwald) [inline src/Utilities/UnitTests/ToolLocationHelper_Tests.cs]: Can you unify the conditions here?

File: src/XMakeBuildEngine/UnitTests/BackEnd/AssemblyTaskFactory_Tests.cs [hunk 5]
```diff
@@ -364,7 +364,7 @@ public void VerifyMatchingUsingTaskParametersDontLaunchTaskHost1()
         /// Verify a good task that does not use the task host can be created when passed task host 
         /// parameters that explicitly match the current process. 
         /// </summary>
-        [Fact(Skip = "Ignored in MSTest")]
+        [Fact]
         // Ignore: Test requires installed toolset.
```
Comment (rainersigwald) [inline src/XMakeBuildEngine/UnitTests/BackEnd/AssemblyTaskFactory_Tests.cs]: Can you remove these comments?

File: src/XMakeTasks/UnitTests/Exec_Tests.cs [hunk 6]
```diff
@@ -85,7 +85,9 @@ public void Timeout()
             Assert.Equal(-1, exec.ExitCode);
             ((MockEngine)exec.BuildEngine).AssertLogContains("MSB5002");
             Assert.Equal(1, ((MockEngine)exec.BuildEngine).Warnings);
-            Assert.Equal(1, ((MockEngine)exec.BuildEngine).Errors);
+
+            // ToolTask does not log an error on timeout.
+            Assert.Equal(0, ((MockEngine)exec.BuildEngine).Errors);
```
Comment (rainersigwald) [inline src/XMakeTasks/UnitTests/Exec_Tests.cs]: Uhh. Do we know _why_? It certainly seems like it should . . .

File: src/XMakeBuildEngine/UnitTests/BackEnd/AssemblyTaskFactory_Tests.cs (hunk 5 above)
Comment (rainersigwald) [inline src/XMakeBuildEngine/UnitTests/BackEnd/AssemblyTaskFactory_Tests.cs]: Lots of these, in multiple files.

**Comment by rainersigwald** [comment]
LGTM (with minor edits).

**Comment by rainersigwald** [comment]
@dotnet-bot test Windows_NT Build please

**Comment by AndyGerlicher** [comment]
I like how there have been 3 failures in a row... 1 build break and 2 different unit tests that I didn't even enable in this PR (they were already enabled). :cry:

**Comment by AndyGerlicher** [comment]
@dotnet-bot test Windows_NT Build please

File: src/XMakeTasks/UnitTests/Exec_Tests.cs (hunk 6 above)
Comment (AndyGerlicher) [inline src/XMakeTasks/UnitTests/Exec_Tests.cs]: Agreed, this one scared me a bit that it's just a warning. I even checked the code and stepped through to be sure. It appears intentional, but I didn't get a chance to see if that changed at some point.

**Comment by AndyGerlicher** [comment]
@dotnet-bot test Windows_NT Build please

**Comment by AndyGerlicher** [comment]
@dotnet-bot test Windows_NT Build please

**Pull request merged**